geode/src/main/java/org/apache/calcite/adapter/geode/rel/GeodeAggregateRel.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.adapter.geode.rel;
+
+import org.apache.calcite.plan.RelOptCluster;
+import org.apache.calcite.plan.RelOptCost;
+import org.apache.calcite.plan.RelOptPlanner;
+import org.apache.calcite.plan.RelTraitSet;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.core.Aggregate;
+import org.apache.calcite.rel.core.AggregateCall;
+import org.apache.calcite.rel.metadata.RelMetadataQuery;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.calcite.util.ImmutableBitSet;
+import org.apache.calcite.util.Util;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Implementation of {@link org.apache.calcite.rel.core.Aggregate} relational expression in Geode.
+ */
+public class GeodeAggregateRel extends Aggregate implements GeodeRel {
+
+  public GeodeAggregateRel(RelOptCluster cluster,
+                           RelTraitSet traitSet,
+                           RelNode child,
+                           boolean indicator,
+                           ImmutableBitSet groupSet,
+                           List<ImmutableBitSet> groupSets,
+                           List<AggregateCall> aggCalls) {
+
+    super(cluster, traitSet, child, indicator, groupSet, groupSets, aggCalls);
+
+    assert getConvention() == GeodeRel.CONVENTION;
+    assert getConvention() == input.getConvention(); //??
```

Comment:
Is there a remaining issue to be resolved with this line? If not, it would be good to remove the `//??` comment.

Reply:
+1